assets/js/app.js

```diff
@@ -0,0 +1,390 @@
+                    type="checkbox"
+                    role="switch"
+                    ${isCoinSelected(id) ? `checked` : ""}
+                  />
+                </div>
+              </div>
+              <p class="card-text">${name}</p>
+              <button 
+                type="button" 
+                class="btn btn-primary" 
+                data-bs-toggle="popover" 
+                data-bs-title="Coin Info" 
+                data-bs-content="<div style='text-align: center;'><img src='${small}' alt='coin image' style='width: 100px; height: 100px;' /><br/>USD Price: ${usd}$ <br/>EUR Price ${eur}€ <br/> ILS Price ${ils}₪</div>">
+                More Details
+              </button>
+            </div>
+          </div>
+        `;
+      } catch (error) {
+        console.error(`Error fetching details for ${id}:`, error);
+        return `
+          <div class="card" id="${id}">
+            <div class="card-body">
+              <div class="switchFlex">
+                <h5 class="card-title">${symbol.toUpperCase()}</h5>
+                <div class="form-check form-switch >
+                  <input
+                  id="${id}check"
+                    class="form-check-input"
+                    type="checkbox"
+                    role="switch"
+                    ${isCoinSelected(id) ? `checked` : ""}
+                  />
+                </div>
+              </div>
+              <p class="card-text">${name}</p>
+              <p class="text-danger">Failed to load additional details</p>
+            </div>
+          </div>
+        `;
+      }
+    });
+
+    const htmlArray = await Promise.all(promises);
+    return htmlArray.join("");
+  };
+
+  const renderAllCoins = (newHTML) => {
+    document.getElementById("coinMain").innerHTML = newHTML;
+    activatePopovers();
+    setupToggleButtons(newHTML);
+  };
+
+  const activatePopovers = () => {
+    const popoverTriggerList = document.querySelectorAll(
+      '[data-bs-toggle="popover"]'
+    );
+    popoverTriggerList.forEach((popoverTriggerEl) => {
+      new bootstrap.Popover(popoverTriggerEl, {
+        html: true,
+      });
+    });
+  };
+
+  const handleSearch = async (searchTerm, allCoins) => {
+    const filteredCoins = allCoins.filter((coin) => {
+      const { name, symbol } = coin;
+      return (
+        name.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        symbol.toLowerCase().includes(searchTerm.toLowerCase())
+      );
+    });
+    if (filteredCoins.length === 0) {
+      const myToast = new bootstrap.Toast(
+        document.getElementById("emptyToast")
+      );
+      myToast.show();
+      showLoader();
+      setTimeout(async () => {
+        const firstHundredCoins = allCoins.slice(0, 100);
+        const newHTML = await getCoinHTML(firstHundredCoins);
+        renderAllCoins(newHTML);
+        hideLoader();
+      }, 1500);
+    }
+    const newHTML = await getCoinHTML(filteredCoins);
+    renderAllCoins(newHTML);
```

Comment:
what happens if this function is called when the selected tab is not the coins tab?